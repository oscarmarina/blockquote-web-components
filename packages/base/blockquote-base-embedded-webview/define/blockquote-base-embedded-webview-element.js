import { BlockquoteBaseEmbeddedWebviewElement } from '../src/BlockquoteBaseEmbeddedWebviewElement.js';

window.customElements.define(
  BlockquoteBaseEmbeddedWebviewElement.is,
  BlockquoteBaseEmbeddedWebviewElement,
);
