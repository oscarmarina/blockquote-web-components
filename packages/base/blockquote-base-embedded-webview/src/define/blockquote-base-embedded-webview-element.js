import {BlockquoteBaseEmbeddedWebviewElement} from '../BlockquoteBaseEmbeddedWebviewElement.js';

window.customElements.define(
  'blockquote-base-embedded-webview-element',
  BlockquoteBaseEmbeddedWebviewElement
);
