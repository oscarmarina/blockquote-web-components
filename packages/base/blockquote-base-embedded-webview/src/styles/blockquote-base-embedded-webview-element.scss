:host,
::slotted([slot="embedded"]) {
  display: block;
  box-sizing: border-box;
  inline-size: 100%;
  block-size: 100%;
  margin: 0;
  border: 0;
}

:host([hidden]),
[hidden] {
  display: none !important;
}

*,
*::before,
*::after {
  box-sizing: inherit;
}
