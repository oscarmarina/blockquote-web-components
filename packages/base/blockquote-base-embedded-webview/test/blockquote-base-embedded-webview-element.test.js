/* eslint-disable import/no-extraneous-dependencies */
import {html, fixture, assert, fixtureCleanup, oneEvent} from '@open-wc/testing';
import '../define/blockquote-base-embedded-webview-element.js';

suite('BlockquoteBaseEmbeddedWebviewElement', () => {
  /**
   * @type {import('../src/index').BlockquoteBaseEmbeddedWebviewElement}
   */
  let el;

  teardown(() => fixtureCleanup());

  suite('Default', () => {
    setup(async () => {
      el = await fixture(html`
        <blockquote-base-embedded-webview-element
          src="test/test.html"
          embedded-title="test"></blockquote-base-embedded-webview-element>
      `);
      await el.updateComplete;
    });

    suite('Semantic Dom and a11y', () => {
      test('SHADOW DOM - Structure test', async () => {
        // @ts-ignore
        await assert.shadowDom.equalSnapshot(el, {ignoreAttributes: ['loading']});
      });

      test('LIGHT DOM - Structure test', async () => {
        // @ts-ignore
        await assert.lightDom.equalSnapshot(el, {ignoreAttributes: ['loading']});
      });

      test('a11y', async () => {
        await assert.isAccessible(el);
      });

      test('element content loaded', async () => {
        el.src = 'test/test2.html';
        const {detail} = await oneEvent(el, 'elementloaded', true);
        assert.isTrue(detail.contentDocument.body.hasAttribute('data-embedded'));
      });

      test('connectedCallback is called only one time', async () => {
        const bodyNode = document.querySelector('body');
        bodyNode?.appendChild(el);
        assert.strictEqual(el.querySelectorAll('iframe').length, 1);
      });

      test('element like object', async () => {
        const elObject = await fixture(html`
          <blockquote-base-embedded-webview-element
            type="object"
            src="test/test.html"
            embedded-title="test"></blockquote-base-embedded-webview-element>
        `);
        const objectNode = elObject.querySelector('object');
        assert.strictEqual(objectNode?.tagName, 'OBJECT');
      });
    });
  });
});
