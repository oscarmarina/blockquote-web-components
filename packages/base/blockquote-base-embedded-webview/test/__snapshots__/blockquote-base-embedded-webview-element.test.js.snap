// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BlockquoteBaseEmbeddedWebviewElement > Default > LIGHT DOM - Structure test > LIGHT DOM 1`] = `
"<blockquote-base-embedded-webview-element
  embedded-title="test"
  src="/test/test.html"
>
  <iframe
    allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
    allowfullscreen=""
    slot="embedded"
    src="/test/test.html"
    style="opacity: 0;"
    title="test"
  >
  </iframe>
</blockquote-base-embedded-webview-element>
"
`;

exports[`BlockquoteBaseEmbeddedWebviewElement > Default > SHADOW DOM - Structure test > SHADOW DOM 1`] = `
"<slot name="embedded">
</slot>
"
`;
